ext/opentelemetry-ext-pymongo/src/opentelemetry/ext/pymongo/__init__.py

```diff
@@ -0,0 +1,108 @@
+
+
+class CommandTracer(monitoring.CommandListener):
+    def __init__(self, tracer):
+        if tracer is None:
+            raise ValueError("The tracer is not provided.")
+        self._tracer = tracer
+        self._span_dict = {}
+
+    def started(self, event: monitoring.CommandStartedEvent):
+        command = event.command.get(event.command_name)
+        if command is None:
+            command = ""
+        name = DATABASE_TYPE + "." + event.command_name + "." + command
+        try:
+            span = self._tracer.start_span(name, kind=SpanKind.CLIENT)
+            span.set_attribute("component", DATABASE_TYPE)
+            span.set_attribute("db.type", DATABASE_TYPE)
+            span.set_attribute("db.instance", event.database_name)
+            span.set_attribute(
+                "db.statement", event.command_name + " " + command
+            )
+            if event.connection_id is not None:
+                span.set_attribute("peer.hostname", event.connection_id[0])
+                span.set_attribute("peer.port", event.connection_id[1])
+
+            # pymongo specific, not specified by spec
+            span.set_attribute("db.mongo.operation_id", event.operation_id)
+            span.set_attribute("db.mongo.request_id", event.request_id)
+
+            for attr in COMMAND_ATTRIBUTES:
+                _attr = event.command.get(attr)
+                if _attr is not None:
+                    span.set_attribute("db.mongo." + attr, str(_attr))
+
+            # Add Span to dictionary
+            self._span_dict[_get_span_dict_key(event)] = span
+        except Exception as ex:  # noqa pylint: disable=broad-except
+            if span is not None:
+                span.set_status(Status(StatusCanonicalCode.INTERNAL, ex))
+                span.end()
+                self._remove_span(event)
+
+    def succeeded(self, event: monitoring.CommandSucceededEvent):
+        span = self._span_dict[_get_span_dict_key(event)]
+        if span is not None:
+            span.set_attribute(
+                "db.mongo.duration_micros", event.duration_micros
+            )
+            span.set_status(Status(StatusCanonicalCode.OK, event.reply))
+            span.end()
+            self._remove_span(event)
+
+    def failed(self, event: monitoring.CommandFailedEvent):
+        span = self._span_dict[_get_span_dict_key(event)]
+        if span is not None:
+            span.set_attribute(
+                "db.mongo.duration_micros", event.duration_micros
+            )
+            span.set_status(Status(StatusCanonicalCode.UNKNOWN, event.failure))
+            span.end()
+            self._remove_span(event)
+
+    def _get_span(self, event):
+        return self._span_dict[_get_span_dict_key(event)]
+
+    def _remove_span(self, event):
+        self._span_dict.pop(_get_span_dict_key(event))
+
+
+def _get_span_dict_key(event):
+    if event.connection_id is not None:
+        return (event.request_id, str(event.connection_id))
```

Comment:
Is there a particular reason to use `str` here?

Reply:
Not needed, removing it